### PR TITLE
Fix deleting profiles (issue #743)

### PR DESCRIPTION
**Pull request by phaumer**
We cannot assume that all API extensions implement all APIs. For example the FTP extension currently only implements USS.  However, the Delete Profile code was doing a request to the API Registry for finding a profile under MVS.  However, all the information that was needed from the return value was the profile type, which is already included in the profile itself. So I just removed the call.

Other minor changes:
- renamed deleteProf() to deleteProfileOnDisk() as the name was not expressing what the method was doing.
- made deletePrompt() and deleteProfileOnDisk() private as they are not used anywhere else.

### CODE REVIEW

**Comment by phaumer** [comment]
I based this PR on PR https://github.com/zowe/vscode-extension-for-zowe/pull/776 as tests were failing in the master branch.

**Comment by jellypuno** [comment]
Note: Merge after #776

**Comment by phaumer** [comment]
> The only thing I noticed were changes in the i18n files after I ran the `vscode:prepublish` script.

I was surprised by that as well and added that commit. The only thing different was that one sting changed position in the file.

**Comment by phaumer** [comment]
@jellypuno it is merge now and I merged into this branch. Looks like all checks are passing as well.

File: src/Profiles.ts [hunk 1]
```diff
@@ -592,6 +553,44 @@ export class Profiles {
...
+        };
+        // confirm that the user really wants to delete
+        if (await vscode.window.showQuickPick([localize("deleteProfile.showQuickPick.yes", "Yes"),
+            localize("deleteProfile.showQuickPick.no", "No")], quickPickOptions) !== localize("deleteProfile.showQuickPick.yes", "Yes")) {
+            this.log.debug(localize("deleteProfile.showQuickPick.log.debug", "User picked no. Cancelling delete of profile"));
+            return;
+        }
+
+        try {
+            this.deleteProfileOnDisk(deletedProfile, profileName, deletedProfile.type);
```
Comment (zFernand0) [inline src/Profiles.ts:572]: If the `deletedProfile` already hast the infromation about the type, why does the `deleteProfileOnDisk` function requires 3 parameters? (is there a `Closure` problem?)

Comment (phaumer) [reply src/Profiles.ts:575]: You are right. I missed that.

Comment (zFernand0) [reply src/Profiles.ts:575]: No worries, I was just wondering if there was a problem accessing the value due to closure issues. So, it's fine 😋

**Pull request merged**